v4/internal/otel_expect.go

```diff
@@ -6,14 +6,66 @@ import (
 
 // OpenTelemetryExpect implements internal.Expect for use in testing.
 type OpenTelemetryExpect struct {
-	*testtrace.Tracer
+	Spans *testtrace.StandardSpanRecorder
 }
 
-// ExpectTxnEvents TODO
-func (e *OpenTelemetryExpect) ExpectTxnEvents(t Validator, want []WantEvent) {}
+func expectSpan(t Validator, want WantSpan, span *testtrace.Span) {
```

Comment:
I'd love your feedback on the implementation of this method.

Basically, I made it so that each field on the `WantSpan` is required. If you specifically want a field to be ignored, you must specify that with the `internal.MatchAnyID` option. An alternative would be to ignore testing of a field if it is found to be an empty string on the `WantSpan`.

Reply:
> An alternative would be to ignore testing of a field if it is found to be an empty string on the WantSpan.

I like this alternative better, thanks for switching to it. It seems more intuitive to me (we only test for what is explicitly specified).